packages/apps/job-launcher/server/src/modules/job/job.service.ts

```diff
@@ -901,18 +913,80 @@ export class JobService {
   }
 
   @Cron(CronExpression.EVERY_10_MINUTES)
-  public async launchCronJob() {
-    this.logger.log('Launch jobs START');
+  public async createEscrowCronJob() {
+    const isCronJobRunning = await this.cronJobService.isCronJobRunning(
+      CronJobType.CreateEscrow,
+    );
+
+    if (isCronJobRunning) {
+      this.logger.log('Previous cron job is not completed yet');
+      return;
+    }
+
+    this.logger.log('Create escrow START');
+    const cronJob = await this.cronJobService.createCronJob(
```

Comment:
does it work if the cron job entity already exists?

Reply:
Yes, that works. Unique index on cron job entity is `cron job type` + `created at`

Reply:
I confirmed multiple cron job entities are created under `create-escrow` type.

Reply:
Why to create multiple? The way I see it, it should have only one entry in the database and we update its status so it works as a flag. For example, if there is no entity or last execution was completed means we can execute it, so when we start update the field updatedat with current date and set completedat to null, this means it is being run. I dont say your implementation is wrong but I think we would be filling a table for each execution, with information that is not actually useful